crates/networking/manager/src/gossipsub/validate/light_client_finality_update.rs

```diff
@@ -0,0 +1,86 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use anyhow::{Ok, anyhow};
+use ream_chain_beacon::beacon_chain::BeaconChain;
+use ream_consensus_misc::constants::beacon::SYNC_COMMITTEE_SIZE;
+use ream_light_client::finality_update::LightClientFinalityUpdate;
+use ream_network_spec::networks::{beacon_network_spec, lean_network_spec};
+use ream_storage::{cache::CachedDB, tables::table::REDBTable};
+
+use crate::gossipsub::validate::result::ValidationResult;
+
+pub async fn validate_light_client_finality_update(
+    update: &LightClientFinalityUpdate,
+    beacon_chain: &BeaconChain,
+    cached_db: &CachedDB,
+) -> anyhow::Result<ValidationResult> {
+    let store = beacon_chain.store.lock().await;
+    let head_root = store.get_head()?;
+    let _state = store
+        .db
+        .state_provider()
+        .get(head_root)?
+        .ok_or_else(|| anyhow!("Could not get beacon state: {head_root}"))?;
+
+    // [IGNORE] The finalized header is greater than that of all previously forwarded finality
+    // updates or it matches the highest previously forwarded slot and also has a supermajority
+    // participation while previously forwarded slot did not indicate supermajority
+    let new_slot = update.finalized_header.beacon.slot;
+    let participation_count = update
+        .sync_aggregate
+        .sync_committee_bits
+        .iter()
+        .filter(|b| *b)
+        .count() as u64;
+
+    let has_supermajority = participation_count * 3 > SYNC_COMMITTEE_SIZE * 2;
+
+    let mut last_forwarded_update = cached_db.seen_forwarded_finality_update_slot.write().await;
+    if let Some(ref mut info) = *last_forwarded_update {
```

Comment:
```suggestion
    if let Some((last_slot, was_last_slot_finalized)) = *last_forwarded_update {
```

Why is this mut? it doesn't look like we are modifying it, also just expand the object, instead of indexing it, do something like I did in my example, you can change the names to something that makes more sense